tests/pytests/test_empty.py

```diff
@@ -0,0 +1,655 @@
+    arr = {
+        'arr': ['a', '', 'c']
+    }
+    arrs = json.dumps(arr, separators=(',', ':'))
+    env.expect('JSON.SET', 'j', '$', arrs).equal('OK')
+    cmd = f'FT.SEARCH jidx isempty(@arr)'.split(' ')
+    expected = [1, 'j', ['$', arrs]]
+    cmd_assert(env, cmd, expected)
+
+    # Empty arrays shouldn't be indexed for this indexing mechanism
+    arr = {
+        'arr': []
+    }
+    arrs = json.dumps(arr, separators=(',', ':'))
+    env.expect('JSON.SET', 'j', '$', arrs).equal('OK')
+    cmd = f'FT.SEARCH jidx isempty(@arr)'.split(' ')
+    expected = EMPTY_RESULT
+    cmd_assert(env, cmd, expected)
+    conn.execute_command('DEL', 'j')
+
+    # Empty object shouldn't be indexed for this indexing mechanism (flatten, [*])
+    obj = {
+        'arr': {}
+    }
+    objs = json.dumps(obj, separators=(',', ':'))
+    env.expect('JSON.SET', 'j', '$', objs).equal('OK')
+    cmd = f'FT.SEARCH jidx isempty(@arr) RETURN 1 arr'.split(' ')
+    expected = EMPTY_RESULT
+    cmd_assert(env, cmd, expected)
+
+    # Searching for emptiness of a non-existing field should return an error
+    obj = {
+        'obj': {}
+    }
+    objs = json.dumps(obj, separators=(',', ':'))
+    env.expect('JSON.SET', 'j', '$', objs).equal('OK')
+    cmd = f'FT.SEARCH jidx isempty(@obj)'.split(' ')
+    expected = EMPTY_RESULT
+    env.expect(*cmd).error().contains('Syntax error: Field not found')
+
+    env.flush()
+
+    # An attempt to index a non-empty object as a TAG (and in general) should fail (coverage)
+    j = {
+        "t": {"lala": "lali"}
+    }
+    js = json.dumps(j)
+    env.expect('FT.CREATE', 'jidx', 'ON', 'JSON', 'SCHEMA', '$.t', 'AS', 't', 'TAG', 'ISEMPTY').ok()
+    env.expect('JSON.SET', 'j', '$', js).equal('OK')
+    cmd = f'FT.SEARCH jidx isempty(@t)'.split(' ')
+    cmd_assert(env, cmd, EMPTY_RESULT)
+
+    # Make sure we experienced an indexing failure, via `FT.INFO`
+    info = index_info(env, 'jidx')
+    env.assertEqual(info['hash_indexing_failures'], 1)
+
+    env.flush()
+
+    # Test that when we index many docs, we find the wanted portion of them upon
+    # empty value indexing
+    env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TAG', 'ISEMPTY').ok()
+    n_docs = 1000
+    for i in range(n_docs):
+        conn.execute_command('HSET', f'h{i}', 't', '' if i % 2 == 0 else f'{i}')
+    res = env.cmd('FT.SEARCH', 'idx', 'isempty(@t)', 'LIMIT', '0', '0')
+    env.assertEqual(int(res[0]), 500)
+
+def testEmptyText():
+    """Tests the indexing and querying of empty TEXT (field type) values"""
+
+    env = Env(moduleArgs="DEFAULT_DIALECT 2")
+    conn = getConnectionByEnv(env)
+
+    def testEmptyTextHash(idx):
+        """Tests the indexing and querying of empty values for a TEXT field of a
+        hash index
+        Extensive tests are added here, specifically to the query part, due to
+        the addition of the `isempty` function syntax added to the parser.
```

Comment:
Consider removing this (would become less relevant as time passes)

Reply:
I would rather it be clear that it was added to the parser here.